src/libpriv/rpmostree-core.c

```diff
@@ -444,6 +423,14 @@ rpmostree_context_new_unprivileged (int basedir_dfd,
   return rpmostree_context_new_internal (basedir_dfd, TRUE, cancellable, error);
 }
 
+/* Use this if no packages will be installed, and we just want a "dummy" run.
+ */
+void
+rpmostree_context_set_is_empty (RpmOstreeContext *self)
```

Comment:
I think we do want the layering core to error out in basically *every* other case besides just regenerating the initramfs locally.  If for example you try to build a container with no packages; another case is that issue where layering a provides of something in the base.  So here the `is_empty` is a flag the caller sets to explicitly set the dummy mode.